html-dialog/html-dialog-demos/src/ch/ivyteam/htmldialog/demo/data/DataSource.java

```diff
@@ -65,28 +65,26 @@ private static boolean filter(Map<String, FilterMeta> filters, Person person) {
     if (filters.isEmpty()) {
       return true;
     }
+    boolean allFiltersMatch = true;
     for (Iterator<String> it = filters.keySet().iterator(); it.hasNext();) {
       String filterProperty = it.next();
-      if (matches(filters, person, filterProperty)) {
-        return true;
-      }
+      String filterValue = filters.get(filterProperty).getFilterValue().toString();
+      allFiltersMatch = allFiltersMatch && matches(filterValue, person, filterProperty);
     }
-    return false;
+    return allFiltersMatch;
   }
 
-  private static boolean matches(Map<String, FilterMeta> filters, Person person, String filterProperty) {
-    String filterValue = filters.get(filterProperty).toString();
```

Comment:
I'm maybe wrong, but instead of all you changes, can't you just change this line here to fix it?
`String filterValue = filters.get(filterProperty).getFilterValue().toString();`

Reply:
no, then the global filter if nothing set has an empty String as filterValue that matches always to true. As a result the column fikter has no effect.
Futher more IMO, severall filters at the same time should be possible. e.g name column AND and birthdate column